pennylane/ops/functions/equal.py

```diff
@@ -72,12 +72,10 @@ def equal(
         >>> qml.equal(op3, op4, check_trainability=False)
         True
     """
+    if op1.arithmetic_depth > 0 or op2.arithmetic_depth > 0:
+        return op1.hash == op2.hash
```

Comment:
I think we should move this block below lines 77-78, instead of having it above.

Reply:
Why?

Reply:
Guess it doesn't matter too much.  We should remove the `op1.arithmetic_depth != op2.arithmetic_depth` check on the next line though.  If if gets to that point, we already know that both arithmetic depths are zero.

Reply:
Done!